### ISSUE TITLE
Were you able to complete this ring?

### ISSUE THREAD

**Issue by Cosmopal** [state-of-the-art/rring#1]
Hey,
The project looks very promising, I would love to colab and get this to completion.

**Comment by rabits** [comment]
Hi @Cosmopal , no unfortunately I was not able to actually implement it. The complexity, as far I remember, was:
* 2.4Ghz antenna design - quite a complicated topic.
* Battery - had a talk to manufacturers and producing of such were pricey
* Vibration device design was a fail since I'm not that great in mini-physics of magnetic fields and without such vibration the ring looks incomplete.

So if you wish to try to actually build it or can somehow help with the issues - that could be a great kick for me to continue the hw & firmware development. For now it's not really needed by anyone - so I stopped development since have some pressing projects.

As far I see I can copy the antenna design of the ring I used as prototype or try to get OmniRing design: https://www.cse.psu.edu/~mkg31/projects/omniring/ , but not quite sure if that will fit... The battery for sure can be used from another cheap smart ring, but about the vibration device I'm still not sure.

**Comment by Cosmopal** [comment]
I think building the ring incrementally will be more practical. We can start small and at least have a working prototype on which we can build up. It's a challenging project for me too, which is why my idea of a first version was very small - basically a counting ring that keeps the count of number of times you touched its touch sensor, and is able to transmit that info. We can modify your current design to include a touch sensor instead of the vibration motor. 

Can you share some details about the three challenges you mentioned above? Let me take a stab at it. 
Meanwhile, we could:
- Replace the BT sensor with NFC, which we can use for both comms and charging
- Leave vibration out of scope if it doesn't work out.

The OmniRing's paper is there, but the design, specs or program is not available. One has to fill a Google form to request info. I did try to reach out to OmniRing team, but they have not been responding. 

A ring is a very potent wearable device, and I'd love to have a tinkerable open source ring.

**Comment by rabits** [comment]
I think it's reasonable to build it incrementally - for sure you can remove some unnecessary parts (just do not fill the places there on the PCB). There is a physical button included in the design, because I found that touch button is not work that great - but if you need you can modify it to touch for sure.

"Replace the BT sensor with NFC" - what do you mean? Not quite sure I understand how nfc will work as a charger and what is "BT sensor".

And what kind of additional info I can provide? Not sure if you looked into the repo - there are 3d models, PCB and sample firmware (just the beginning of it), so it contains everything I have for now. And I recalled that I managed to replace the pcb antenna with solid state one, so one problem less!

**Comment by Cosmopal** [comment]
Yes, I found the docs you had put in the repo. I’m still going through them. I was looking for details like:
- which battery vendors you talked to and what price range did you find?
- What exactly was the challenge with the vibration motor? Was the magnetism from it causing issues with circuit?

**Comment by rabits** [comment]
* About batteries I talked to:
   * Shenzhen Ykxbattery Energy Co., Ltd., they've requested MOQ 20000pcs and sample making cost is $650.
   * Shenzhen Cowon Technology Limited and General Electronics Technology Co., Ltd., but their batteries was not "arc" type
* Vibration device - even the smallest motors are actually don't fit into the form-factor, so I thought about making a custom magnetic-induction one, where magnet mass will be moved in a chamber between 2 points back and forth hitting the walls and making vibration this way: [vibro_device-mk1.blend](https://github.com/state-of-the-art/rring/blob/main/model/vibro_device-mk1.blend). I tested a couple of implementations, but they were working not well...
   * First iteration with the small magnet (1h2d) - incorrect magnetic field:
   <img src="https://github.com/state-of-the-art/rring/assets/1093323/c64b4362-4553-4fa3-93db-cdc975dbf3f4" height="100"> <img src="https://github.com/state-of-the-art/rring/assets/1093323/7b39bb6b-af96-428b-8a45-e24025f1b3ad" height="100"> <img src="https://github.com/state-of-the-art/rring/assets/1093323/9f97190f-5d06-438c-9eca-658152129ec7" height="100"> <img src="https://github.com/state-of-the-art/rring/assets/1093323/92805575-e33d-4386-b2f5-010273c01f9e)" height="100"> <img src="https://github.com/state-of-the-art/rring/assets/1093323/22404376-adcb-4787-a00a-3ac0fbe37902" height="100"> <img src="https://github.com/state-of-the-art/rring/assets/1093323/4770fdcb-6ec6-444d-b3b3-811a968ef123" height="100">
   * Second iteration was with ferromagnetic metal as mass - no pics, the mass moved around but did not made any impact and easily stuck.

**Comment by QuartzLibrary** [comment]
Hi there! Just found this while looking up OS ring designs.

A couple of thoughts in case they are helpful:
- Things like batteries could be sourced by cannibalizing cheap rings from Aliexpress or similar electronics. That has its own challenges, but could allow you to get a prototype up and running.
- Random thought on vibration is that if tactile feedback is what's desired, then maybe having the ring exert a light pressure on the finger is sufficient (and less energy intensive?). This can still allow for waterproofing if the 'piston' is moved from behind a wall of plastic. Still not as self-contained though.
- While on the topic of whacky ideas, a very small Peltier module could also in theory provide physical feedback.
- I like the idea of inobtrusive wearables, but ideally I would like them to be even more set-and-forget than rings. A design that looks more like blood-glucose monitors, might be easier to design and very cheap to produce. Think a small disk you stick to your arm, or ankle, or whatever really, and forget about for the next two weeks or more.
I was looking for OS rings designs to see how hard it would be to repurpose them for this, as it seems much easier.

**Comment by nmn28** [comment]
hi guys. this is the battery that oura ring uses: 3.7V 16mAh Lithium Battery - Grepow YE160723G. its custom curved lipo.

would love to connect further

**Comment by ahmetselmanerdogan** [comment]
There are deficiencies, I would like to talk about the details of the file, can you contact me?

**Comment by rabits** [comment]
@ahmetselmanerdogan why not to discuss those right here? It's open conversation - nothing to hide, especially deficiencies.

**Comment by ahmetselmanerdogan** [comment]
> @ahmetselmanerdogan why not to discuss those right here? It's open conversation - nothing to hide, especially deficiencies.

You are right, but how could I reach you? Can you write me on wp to meet? My number is +90 5078872959

**Comment by zehaaan** [comment]
Any updates... I am all in this project...  i wanna add ai features here....

**Comment by rabits** [comment]
Hi @zehaaan , unfortunately project is abandoned, because there are cheap & convenient hardware existing already and even with a way to flash your own firmware: https://github.com/atc1441/ATC_SR08_Ring/ - it's using better chip and almost as good as my project, except it doesn't have physical button (touch one) and vibromotor - but that's small things.

**Comment by SirNate0** [comment]
Another open source ring: https://github.com/stawiski/open-ring

Which has some sort of hardware made, I'm a little unclear on exactly what. You can follow the creator's blog [here](https://interrupt.memfault.com/blog/smart-ring-development-part-2).
